clientv3/auth.go

```diff
@@ -85,3 +90,23 @@ func (auth *auth) RoleAdd(ctx context.Context, name string) (*AuthRoleAddRespons
 	resp, err := auth.remote.RoleAdd(ctx, &pb.AuthRoleAddRequest{Name: name})
 	return (*AuthRoleAddResponse)(resp), err
 }
+
+func (auth *auth) RoleGrant(ctx context.Context, name string, key string, readPerm bool, writePerm bool) (*AuthRoleGrantResponse, error) {
```

Comment:
like remove readPerm bool and wrtiePerm bool and add a permType. So we can get rid of the switch at line 97 and simplify this method signature.